server/getTeamsData.go

```diff
@@ -132,6 +132,89 @@ func (p *Plugin) GetMSTeamsTeamChannels(teamID, userID string) ([]msteams.Channe
 	return channels, http.StatusOK, nil
 }
 
+func (p *Plugin) LinkChannels(userID, mattermostTeamID, mattermostChannelID, msTeamsTeamID, msTeamsChannelID string) (string, int) {
+	channel, appErr := p.API.GetChannel(mattermostChannelID)
+	if appErr != nil {
+		p.API.LogError("Unable to get the current channel information.", "ChannelID", mattermostChannelID, "Error", appErr.Message)
+		return "Unable to get the current channel information.", http.StatusInternalServerError
+	}
+
+	if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+		p.API.LogError("Linking/unlinking a direct or group message is not allowed.", "ChannelType", channel.Type)
+		return "Linking/unlinking a direct or group message is not allowed", http.StatusForbidden
+	}
+
+	if !p.API.HasPermissionToChannel(userID, mattermostChannelID, model.PermissionManageChannelRoles) {
+		p.API.LogError("Unable to link the channel. You have to be a channel admin to link it.", "ChannelID", mattermostChannelID)
+		return "Unable to link the channel. You have to be a channel admin to link it.", http.StatusForbidden
+	}
+
+	if !p.store.CheckEnabledTeamByTeamID(mattermostTeamID) {
+		p.API.LogError("This team is not enabled for MS Teams sync.", "TeamID", mattermostTeamID)
+		return "This team is not enabled for MS Teams sync.", http.StatusForbidden
+	}
+
+	link, err := p.store.GetLinkByChannelID(mattermostChannelID)
+	if err != nil {
+		p.API.LogError("Error occurred while getting channel link with Mattermost channelID.", "ChannelID", mattermostChannelID, "Error", err.Error())
+		return "Error occurred while getting channel link with Mattermost channelID.", http.StatusInternalServerError
+	}
+	if link != nil {
+		return "A link for this channel already exists. Please unlink the channel before you link again with another channel.", http.StatusBadRequest
```

Comment:
Please unlink the channel first before linking again

Reply:
fix this everywhere

Reply:
I don't think what we have is fine